tests/functional/conftest.py

```diff
@@ -1,138 +1,158 @@
-    """
-    async def _token(
-        grant_type: str = "authorization_code",
-        test_app: ApigeeApiDeveloperApps = None,
-        **kwargs
-    ):
-        if test_app:
-            # Use provided test app
-            oauth = OauthHelper(test_app.client_id, test_app.client_secret, test_app.callback_url)
-            resp = await oauth.get_token_response(grant_type=grant_type, **kwargs)
-        else:
-            # Use env vars
-            oauth = OauthHelper(
-                client_id=config.CLIENT_ID,
-                client_secret=config.CLIENT_SECRET,
-                redirect_uri="https://example.org/callback",
-            )
-            resp = await oauth.get_token_response(grant_type=grant_type, **kwargs)
-
-        if resp['status_code'] != 200:
-            message = 'unable to get token'
-            raise RuntimeError(f"\n{'*' * len(message)}\n"
-                               f"MESSAGE: {message}\n"
-                               f"URL: {resp.get('url')}\n"
-                               f"STATUS CODE: {resp.get('status_code')}\n"
-                               f"RESPONSE: {resp.get('body')}\n"
-                               f"HEADERS: {resp.get('headers')}\n"
-                               f"{'*' * len(message)}\n")
-        return resp['body']
-
-    return _token
+    return product
 
 
 @pytest.fixture(scope="function")
-async def setup_session(request):
+def setup_session(request, _jwt_keys, apigee_environment, client, api_products):
     """This fixture is called at a function level.
     The default app created here should be modified by your tests.
     """
 
-    product = await _product_with_full_access()
+    product = _product_with_full_access(api_products)
+    product.update_environments([config.ENVIRONMENT], api_products=api_products)
+
+    LOGGER.info(f'product.proxies: {product.proxies}')
+
     print("\nCreating Default App..")
     # Create a new app
+    developer_apps = DeveloperAppsAPI(client=client)
+
     app = ApigeeApiDeveloperApps()
-    await app.create_new_app(
-        callback_url="https://example.org/callback"
+    create_app_response = app.create_new_app(
+        callback_url="https://example.org/callback",
+        status="approved",
+        jwks_resource_url=config.JWKS_RESOURCE_URL,
+        products=[product.name],
+        developer_apps=developer_apps
     )
-    # Assign the new product to the app
-    await app.add_api_product([product.name])
 
-    # Set default JWT Testing resource url
-    await app.set_custom_attributes(
-            {
-                'jwks-resource-url': config.JWKS_RESOURCE_URL
-            }
+    LOGGER.info(f'create_app_response: {create_app_response}')
+
+    # Set up app config
+    client_credentials_config = ClientCredentialsConfig(
+        environment=apigee_environment,
+        identity_service_base_url=f"https://{apigee_environment}.api.service.nhs.uk/{config.OAUTH_PROXY}",
```

Comment:
Similar to above, looks like the default value for `ClientCredentialsConfig.indentity_service_base_url` is the `oauth-mock2`. Is it worth removing this to keep things cleaner? Which would mean we could remove the added line in `config`.

Reply:
As above, worth doing under SPINEDEM-1680